website/docs/getting-started/reasoning-routing-quickstart.md

```diff
@@ -0,0 +1,205 @@
+# Reasoning Routing Quickstart
+
+This short guide shows how to enable and verify “reasoning routing” in the Semantic Router:
+- Minimal config.yaml fields you need
+- Example request/response (OpenAI-compatible)
+- A comprehensive evaluation command you can run
+
+Prerequisites
+- A running OpenAI-compatible backend for your models (e.g., vLLM or any OpenAI-compatible server). It must be reachable at the addresses you configure under vllm_endpoints (address:port).
+- Envoy + the router (see Start the router section)
+
+1) Minimal configuration
+Put this in config/config.yaml (or merge into your existing config). It defines:
+- Categories that require reasoning (e.g., math)
+- Reasoning families for model syntax differences (DeepSeek/Qwen3 use chat_template_kwargs; GPT-OSS/GPT use reasoning_effort)
+- Which concrete models use which reasoning family
+- The classifier (required for category detection; without it, reasoning will not be enabled)
+
+```yaml
+# Category classifier (required for reasoning to trigger)
+classifier:
+  category_model:
+    model_id: "models/category_classifier_modernbert-base_model"
+    use_modernbert: true
+    threshold: 0.6
+    use_cpu: true
+    category_mapping_path: "models/category_classifier_modernbert-base_model/category_mapping.json"
+
+# vLLM endpoints that host your models
+vllm_endpoints:
+  - name: "endpoint1"
+    address: "127.0.0.1"
+    port: 8000
+    models: ["deepseek-v3", "qwen3-7b", "openai/gpt-oss-20b"]
+    weight: 1
+
+# Reasoning family configurations (how to express reasoning for a family)
+reasoning_families:
+  deepseek:
+    type: "chat_template_kwargs"
+    parameter: "thinking"
+  qwen3:
+    type: "chat_template_kwargs"
+    parameter: "enable_thinking"
+  gpt-oss:
+    type: "reasoning_effort"
+    parameter: "reasoning_effort"
+  gpt:
+    type: "reasoning_effort"
+    parameter: "reasoning_effort"
+
+# Default effort used when a category doesn’t specify one
+default_reasoning_effort: medium  # low | medium | high
+
+# Map concrete model names to a reasoning family
+model_config:
+  "deepseek-v3":
+    reasoning_family: "deepseek"
+    preferred_endpoints: ["endpoint1"]
+  "qwen3-7b":
+    reasoning_family: "qwen3"
+    preferred_endpoints: ["endpoint1"]
+  "openai/gpt-oss-20b":
+    reasoning_family: "gpt-oss"
+    preferred_endpoints: ["endpoint1"]
+
+# Categories: which kinds of queries require reasoning and at what effort
+categories:
+- name: math
+  use_reasoning: true
+  reasoning_effort: high  # overrides default_reasoning_effort
+  reasoning_description: "Mathematical problems require step-by-step reasoning"
+  model_scores:
+  - model: openai/gpt-oss-20b
+    score: 1.0
+  - model: deepseek-v3
+    score: 0.8
+  - model: qwen3-7b
+    score: 0.8
+
+- name: general
```

Comment:
at the moment, all the categories must be from mmlu-pro. There is no `general` there. You can create an issue to support generic, free style categories and we can map the mmlu-pro categories to them.